### PR TITLE
Clarified Terms of Use

### PR DESCRIPTION
**Pull request by David-Chadwick**
This updates #146 (which can now be deleted). It has addressed the comments made on #146

***
<a href="https://pr-preview.s3.amazonaws.com/David-Chadwick/vc-data-model/pull/170.html" title="Last updated on May 29, 2018, 10:06 AM GMT (0bfd396)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/170/9ea3a8d...David-Chadwick:0bfd396.html" title="Last updated on May 29, 2018, 10:06 AM GMT (0bfd396)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -924,8 +924,11 @@ <h2>Terms of Use</h2>
         <dl>
           <dt><var>termsOfUse</var></dt>
           <dd>The value of this property MUST be one or more terms of use
-            descriptions that provide enough information to a <a>verifier</a>
+            that provide enough information to a <a>verifier</a>
             to determine how they may utilize the given information.
+            Each terms of use is uniquely identified by a type and id. The precise
```
Comment (msporny) [inline index.html]: While `id` is preferred, it is not required. Nodes in the graph that are not assigned an `id` have one auto-generated for them during canonicalization or JSON-LD processing. We should make sure that people know that `id` is optional via a SHOULD. Also, keep in mind that the ODRL community will probably have more than a few thoughts on this as well.

Comment (David-Chadwick) [reply index.html]: @msporny . When you say id is not required, this may be the case for pure JSON-LD. But in the VC spec we can mandate the use of the id if we want to. In other words, the VC spec can place constraints on JSON-LD when it is deemed to be appropriate. For example, a VC must itself always have a type and id property.

Comment (msporny) [reply index.html]: > But in the VC spec we can mandate the use of the id if we want to.

Yes, I'm stating that I don't think id is necessary here because you're describing potentially transient information and thus referring to it by `id` is something we're forcing developers to do when they will end up just creating a random throw-away identifier (like a UUID).

Comment (stonematt) [reply index.html]: 1. can ToU "attach" (using that term loosely) to any of our artifacts: Verifiable Credential or Presentation?
1. can ToU be added by either Issuer or Holder?
1. is the "owner/setter" of the ToU clear?

File: index.html [hunk 2]
```diff
@@ -917,15 +917,29 @@ <h2>Extensibility</h2>
         <h2>Terms of Use</h2>
         <p>Terms of use can be utilized by an <a>issuer</a>, <a>subject</a> or a
         <a>holder</a> to express limitations on the use of information
-        expressed by the Verifiable Credentials Model. The expression of
+        expressed by the Verifiable Credentials Model. The issuer places the terms of use
+       inside the credential before it is converted into a verifiable credential.
+       The holder places the terms of use inside the <presentation> before it is converted
```
Comment (davidlehn) [inline index.html]: `<presentation>` is being parsed as an unknown tag and dropped.  Escape the angle brackets if that's what you meant to do.  Or maybe it's supposed to be `<a>presentation</a>` and a definition added to terms.html?

File: index.html [hunk 3]
```diff
@@ -917,15 +917,29 @@ <h2>Extensibility</h2>
...
+       into a verifiable presentation.
+
+Note: it is for further study how a subject who is not a holder places terms of use on his
```
[the remark this answers is not in the document]
Comment (David-Chadwick) [reply index.html]: done, thanks

File: index.html (hunk 2 above)
Comment (David-Chadwick) [reply index.html]: I was prejudging profile moving to presentation, but got the syntax wrong. But I had better revert to profile for now

**Comment by David-Chadwick** [comment]
@msporny The id is now optional as you requested

File: index.html [hunk 4]
```diff
@@ -917,15 +917,30 @@ <h2>Extensibility</h2>
...
+        terms of use may be performed via the following <a>property</a>:
         </p>
 
         <dl>
           <dt><var>termsOfUse</var></dt>
           <dd>The value of this property MUST be one or more terms of use
-            descriptions that provide enough information to a <a>verifier</a>
-            to determine how they may utilize the given information.
+            that provide enough information to a <a>verifier</a>
+            to determine how they MUST utilize the given information.
```
[the remark this answers is not in the document]
Comment (David-Chadwick) [reply index.html]: The first MUST is defining what the content of the VC must contain, whilst the second MUST is saying how the verifier must behave. So they are both needed as they apply to different things.
It is possible to test the second must by carefully crafting a terms of  use that can be externally measured. E.g. it could be 'transaction must be less than $100', then two transactions are sent to the verifier using the same VC, one for <100 and one >100.  The second one must be rejected whilst the first could be accepted or rejected. But one could think of lots of others as well e.g. Subject must be notified by email when xxx happens, then perform a test that does xxx and check that the subject receives an email.

File: index.html [hunk 5]
```diff
@@ -917,15 +917,30 @@ <h2>Extensibility</h2>
...
+            Each termsOfUse comprises its type, for example, IssuerPolicy, and
```
[the remark this answers is not in the document]
Comment (David-Chadwick) [reply index.html]: done

File: index.html [hunk 6]
```diff
@@ -917,15 +917,30 @@ <h2>Extensibility</h2>
...
         <p>Terms of use can be utilized by an <a>issuer</a>, <a>subject</a> or a
         <a>holder</a> to express limitations on the use of information
-        expressed by the Verifiable Credentials Model. The expression of
-        terms of use are performed via the following <a>property</a>:
+        expressed by the Verifiable Credentials Model. The <a>issuer</a> places 
+       the terms of use
+       inside the credential before it is converted into a verifiable credential.
+       The <a>holder</a> places the terms of use inside the <a>profile</a> before it is
+       converted into a <a>verifiable profile</a>.</p>
+	<p>
```
[the remark this answers is not in the document]
Comment (David-Chadwick) [reply index.html]: Turned it class note.

File: index.html [hunk 7]
```diff
@@ -917,15 +917,30 @@ <h2>Extensibility</h2>
         <h2>Terms of Use</h2>
         <p>Terms of use can be utilized by an <a>issuer</a>, <a>subject</a> or a
         <a>holder</a> to express limitations on the use of information
-        expressed by the Verifiable Credentials Model. The expression of
-        terms of use are performed via the following <a>property</a>:
+        expressed by the Verifiable Credentials Model. The <a>issuer</a> places 
+       the terms of use
+       inside the credential before it is converted into a verifiable credential.
```
[the remark this answers is not in the document]
Comment (David-Chadwick) [reply index.html]: Done

File: index.html [hunk 8]
```diff
@@ -946,8 +961,8 @@ <h2>Terms of Use</h2>
     "ageOver": 21
   },
   <span class="highlight">"termsOfUse": [{
-    "type": "Policy",
```
[the remark this answers is not in the document]
Comment (David-Chadwick) [reply index.html]: This is only an example, and we do not state that it conforms to ODRL, so it does not need to. Furthermore the above issue says we are still exploring this.

File: index.html [hunk 9]
```diff
@@ -965,38 +980,40 @@ <h2>Terms of Use</h2>
...
+	}],
+	"proof": { ...}
 }
         </pre>
 
         <p>
-In the example above, the <a>holder</a> is prohibiting a <a>verifier</a> from
-using the information provided to correlate the <a>holder</a> using a
+In the example above, the <a>holder</a>, who is also the <a>subject</a> 
+is prohibiting the <a>verifier</a> (https://wineonline.example.org) from
```
[the remark this answers is not in the document]
Comment (David-Chadwick) [reply index.html]: done

File: index.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (burnburn) [reply index.html]: At the risk of stepping into a long conversation without the full background, I will comment that the way in which this MUST is worded is not testable.  To see this, try to reword the MUST above in the form 'The verifier MUST xyz'.  I would suggest that this MUST be changed to 'are to'.  You can then provide additional normative MUST statements as necessary, rather than as part of a parenthetical.  My comment is purely structural, not an attempt to read on whether normative statements are appropriate.

[the remark this answers is not in the document]
Comment (David-Chadwick) [reply index.html]: @burnburn How about this wording
 "The value of this property MUST be one or more terms of use that tell the verifier what actions it MUST perform if it is to accept the verifiable credential or verifiable profile. If the verifier is not willing to accept these terms of use then it MUST reject the verifiable credential or verifiable profile."

**Comment by David-Chadwick** [comment]
This can now be deleted as it has been superceded by PR#188 which is a rebased version of this.

**Comment by stonematt** [comment]
Closed based on @David-Chadwick comment above, replaced by PR #188
